File: lanaya-business/src/main/java/cn/lanaya/business/service/CommodityService.java
package cn.lanaya.business.service;

import cn.lanaya.business.entity.Commodity;
import cn.lanaya.business.vo.CommodityVO;
import cn.lanaya.common.bean.PageQO;
import com.github.pagehelper.PageInfo;
import java.util.List;

public interface CommodityService {
    int countByClause(CommodityVO record);

    int deleteByClause(CommodityVO record);

    int deleteByPrimaryKey(Long id);

    int insert(Commodity record);

    int insertSelective(Commodity record);

    List<Commodity> selectByClause(Commodity record);

    Commodity selectByPrimaryKey(Long id);

    int updateByClauseSelective(Commodity entity, CommodityVO param);

    int updateByPrimaryKeySelective(Commodity record);

    int updateByPrimaryKey(Commodity record);

    PageInfo<CommodityVO> pageByClause(PageQO page);

    int deleteByPrimaryKeyBatch(List<CommodityVO> list);

    int insertSelectiveBatch(List<Commodity> list);
}

File: lanaya-business/src/main/resources/static/js/service/commodityService.js
app.service('commodityService', function ($http) {
    this.pageByClause = function (entity) {
        return $http.post('../commodity/pageByClause', entity);
    };

    this.insertSelective = function (entity) {
        return $http.post('../commodity/insertSelective', entity);
    };

    this.updateByPrimaryKeySelective = function () {

        return $http.post('../commodity/updateByPrimaryKeySelective', $scope.entity);
    };

    this.selectByPrimaryKey = function (id) {
        return $http.post('../commodity/selectByPrimaryKey', json({id:id}));
    };
    this.deleteByPrimaryKeyBatch = function () {
        return $http.post('../commodity/deleteByPrimaryKeyBatch', '');
    };
});